src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/BotAction.cs

```diff
@@ -15,10 +17,23 @@ public class BotAction
 
         public BotActionType type;
         public IBotActionData data;
-        public bool IsCompleted => data.IsCompleted();
+        public bool? IsCompleted => data.IsCompleted(); // returns null if this action runs until the keyframecriteria are met
 
         public int EffectiveApiVersion => Math.Max(apiVersion, data?.EffectiveApiVersion() ?? 0);
 
+        // Called before the first call to ProcessAction to allow data setup by the action code
+        public void StartAction(int segmentNumber, Dictionary<int, TransformStatus> currentUITransforms, Dictionary<int, TransformStatus> currentGameObjectTransforms)
+        {
+            data.StartAction(segmentNumber, currentUITransforms, currentGameObjectTransforms);
+        }
+
+        // called at least once per frame
+        // bool firstCall tracks if this is the first call for this botaction
```

Comment:
Is this an old comment? I don't see firstCall anywhere

Reply:
removed/updated